File: non-native/master_announcer/lib/repository/repository.dart

import 'package:master_announcer/domain/announcement.dart';
import 'package:master_announcer/repository/database.dart';
import 'dart:async';
import 'package:path/path.dart';
import 'package:sqflite/sqflite.dart';


class Repository{

    AnnouncementDatabase announcementDatabase;

    static final _repository=Repository._internal();

    Repository._internal(){
      announcementDatabase=AnnouncementDatabase();
    }

    factory Repository(){
      return _repository;
    }

   Future<int> addAnnouncement(Announcement a) async {

      return await announcementDatabase.addAnnouncement(a);
    }

    Future<int> addAnnouncementId(Announcement a) async {

      return await announcementDatabase.addAnnouncementId(a);
    }

    Future<List<Announcement>> getAnnouncements() async {
      List<Announcement> a=await announcementDatabase.getAnnouncements();
      return a;
    }


    Future deleteAnnouncement(int id) async {
      await announcementDatabase.deleteAnnouncement(id);
    }

    Future editAnnouncement(Announcement newAnnouncement) async {
      await announcementDatabase.updateAnnouncement(newAnnouncement);
    }

    Future addToAdd(int id)async{
      await announcementDatabase.insertToAdd(id);
    }

    Future deleteAll()async{
      await announcementDatabase.deleteAll();
      await announcementDatabase.deleteToAdd();
    }

    Future<List<int>> getToAdd() async{
      return await announcementDatabase.getToAdd();
    }
}

File: native/app/src/main/java/com/example/masterannouncer/repository/Repository.kt
package com.example.masterannouncer.repository

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.example.masterannouncer.domain.Announcement
import com.example.masterannouncer.domain.AnnouncementDAO
import com.example.masterannouncer.domain.AnnouncementId
import java.util.stream.Collectors

class Repository(private val roomDAO: AnnouncementDAO) {

    val announcements: LiveData<List<Announcement>> = roomDAO.getAnnouncements()


    suspend fun addAnnouncement(announcement: Announcement){
        roomDAO.insert(announcement)
    }

    suspend fun editAnnouncement(announcement: Announcement){
        roomDAO.updateAnnouncement(announcement)
    }


    suspend fun deleteAnnouncement(announcement: Announcement){
        roomDAO.deleteAnnouncement(announcement)
    }

    suspend fun deleteAll(){
        roomDAO.deleteAll();
    }

    suspend fun getToAdd(): List<AnnouncementId>{
        return roomDAO.getAnnouncementsToSync()
    }

    suspend fun insertToAdd(id:Long){
        return roomDAO.insertToAdd(AnnouncementId(id))
    }

    suspend fun deleteAllToAdd(){
        roomDAO.deleteAllToAdd()
    }

    suspend fun getAnnouncementsStatic():List<Announcement>{
        return roomDAO.getAnnouncementsStatic()
    }
}